.github/workflows/tests.yml

```diff
@@ -12,25 +12,45 @@ concurrency:
 
 jobs:
   setup_for_tests:
-    name: Get test projects to run
-    runs-on: ubuntu-latest
+    name: Enumerate test projects (${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+        matrix:
+          include:
+            - os: ubuntu-latest
+              dotnet_script: ./dotnet.sh
+            - os: windows-latest
+              dotnet_script: .\dotnet.cmd
     outputs:
       tests_matrix: ${{ steps.generate_test_matrix.outputs.tests_matrix }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
+      - name: Get list of tests
+        env:
+          CI: false
+        run: >
```

Comment:
for my learning, why do we sometimes use `|` and sometimes use `>`?

Reply:
I was using `|` earlier as just continuing what was being used. `|` means that it keeps the newlines, so:
```
- run: |
     foo
     bar
```

.. would be `foo`, and `bar` as separate commands on the shell. And for unix we were using `\` at the end of the lines to concatenate them. But that doesn't work on windows, so I started using `>` - which by default converts the newlines to spaces, so the lines join to become one.

https://yaml-multiline.info